scripts/buildorder.py

```diff
@@ -262,104 +278,133 @@ def read_packages_from_directories(directories, fast_build_mode, full_buildmode)
-    leaf_pkgs = [pkg for pkg in pkgs_map.values() if not pkg.deps]
-
-    if not leaf_pkgs:
-        die('No package without dependencies - where to start?')
-
-    # Sort alphabetically:
-    pkg_queue = sorted(leaf_pkgs, key=lambda p: p.name)
-
-    # Topological sorting
-    visited = set()
-
-    # Tracks non-visited deps for each package
-    remaining_deps = {}
-    for name, pkg in pkgs_map.items():
-        remaining_deps[name] = set(pkg.deps)
-        for subpkg in pkg.subpkgs:
-            remaining_deps[subpkg.name] = set(subpkg.deps)
-
-    while pkg_queue:
-        pkg = pkg_queue.pop(0)
-        if pkg.name in visited:
-            continue
-
-        # print("Processing {}:".format(pkg.name), pkg.needed_by)
-        visited.add(pkg.name)
-        build_order.append(pkg)
-
-        for other_pkg in sorted(pkg.needed_by, key=lambda p: p.name):
-            # Remove this pkg from deps
-            remaining_deps[other_pkg.name].discard(pkg.name)
-            # ... and all its subpackages
-            remaining_deps[other_pkg.name].difference_update(
-                [subpkg.name for subpkg in pkg.subpkgs]
-            )
-
-            if not remaining_deps[other_pkg.name]:  # all deps were already appended?
-                pkg_queue.append(other_pkg)  # should be processed
-
-    if set(pkgs_map.values()) != set(build_order):
-        print("ERROR: Cycle exists. Remaining: ", file=sys.stderr)
-        for name, pkg in pkgs_map.items():
-            if pkg not in build_order:
-                print(name, remaining_deps[name], file=sys.stderr)
-
-        # Print cycles so we have some idea where to start fixing this.
-        def find_cycles(deps, pkg, path):
-            """Yield every dependency path containing a cycle."""
-            if pkg in path:
-                yield path + [pkg]
-            else:
-                for dep in deps[pkg]:
-                    yield from find_cycles(deps, dep, path + [pkg])
+def remove_only_installing_deps(pkgs_map, deps):
+    """Complete replacement of packages that have the `TERMUX_{SUB}PKG_ONLY_INSTALLING` (`only_installing` value)
+    variable set to `true` with their dependencies from the `pkg_deps` value."""
 
-        cycles = set()
-        for pkg in remaining_deps:
-            for path_with_cycle in find_cycles(remaining_deps, pkg, []):
-                # Cut the path down to just the cycle.
-                cycle_start = path_with_cycle.index(path_with_cycle[-1])
-                cycles.add(tuple(path_with_cycle[cycle_start:]))
-        for cycle in sorted(cycles):
-            print(f"cycle: {' -> '.join(cycle)}", file=sys.stderr)
+    # list of packages that have been removed from the dependency list,
+    # it is necessary that these packages cannot appear in the dependency list again
+    pkgs_only_installing = set()
 
-        sys.exit(1)
+    while True:
+        boots_only_installing = [pkgs_map[dep].only_installing for dep in deps]
```

Comment:
What does `boots` mean here?
If it's an abbreviation for `bootstrap`, please just write it out.

Reply:
I accidentally put the letter `t` instead of `l`, the correct variable is `bools_only_installing`.

Reply:
Okay, so what's that do?
It looks like it's a list(?) of bools then?

Reply:
> It looks like it's a list(?) of bools then?

Yes

> Okay, so what's that do?

This list is used to check if there are packages with the `only_installing` variable set to `True`. And also to get these packages through the index of this list.

Reply:
Okay yeah, that makes sense.
Still think it could probably use a better name.
Not that I can come up with a better one on the spot.

Reply:
What about `pkgs_bool`?

Reply:
Eh `bools_only_installing` is probably fine.

Reply:
ok